modin/experimental/core/execution/native/implementations/hdk_on_native/partitioning/partition_manager.py

```diff
@@ -66,23 +66,12 @@ def from_pandas(cls, df, return_dims=False, encode_col_names=True):
             Tuple holding array of partitions, list of columns with unsupported
             data and optionally partitions' dimensions.
         """
-        at, unsupported_cols = cls._get_unsupported_cols(df)
-
-        if len(unsupported_cols) > 0:
-            # Putting pandas frame into partitions instead of arrow table, because we know
-            # that all of operations with this frame will be default to pandas and don't want
-            # unnecessaries conversion pandas->arrow->pandas
-            parts = [[cls._partition_class(df)]]
-            if not return_dims:
-                return np.array(parts), unsupported_cols
-            else:
-                row_lengths = [len(df)]
-                col_widths = [len(df.columns)]
-                return np.array(parts), row_lengths, col_widths, unsupported_cols
+        unsupported_cols = cls._get_unsupported_cols(df)
+        parts = np.array([[cls._partition_class(df)]])
+        if not return_dims:
+            return parts, unsupported_cols
         else:
-            # Since we already have arrow table, putting it into partitions instead
-            # of pandas frame, to skip that phase when we will be putting our frame to HDK
-            return cls.from_arrow(at, return_dims, unsupported_cols, encode_col_names)
+            return parts, [len(df)], [len(df.columns)], unsupported_cols
```

Comment:
Why are dimensions returned in lists?
```suggestion
            return parts, len(df), len(df.columns), unsupported_cols
```

Reply:
To comply with the parent method which returns lists of dimensions for each partition.